commit 9d9ab00e672b
Author: Kyle Machulis <kyle@nonpolynomial.com>
Date:   Mon Jun 25 16:59:08 2018 -0700

    Bug 1469879 - Add check for closed socket on Multicast IPC parent funcs. r=baku, a=lizzard
    
    MozReview-Commit-ID: CaNYPtgFz87
    
    --HG--
    extra : source : ca98b6f47b4e33673291a4dc2a4ebca28ce6db00
---
 dom/network/UDPSocketParent.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git dom/network/UDPSocketParent.cpp dom/network/UDPSocketParent.cpp
index 5198c4be50c06..174d1e87b13e1 100644
--- dom/network/UDPSocketParent.cpp
+++ dom/network/UDPSocketParent.cpp
@@ -450,6 +450,12 @@ mozilla::ipc::IPCResult
 UDPSocketParent::RecvJoinMulticast(const nsCString& aMulticastAddress,
                                    const nsCString& aInterface)
 {
+  if (!mSocket) {
+    NS_WARNING("multicast socket is closed");
+    FireInternalError(__LINE__);
+    return IPC_OK();
+  }
+
   nsresult rv = mSocket->JoinMulticast(aMulticastAddress, aInterface);
 
   if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -463,6 +469,12 @@ mozilla::ipc::IPCResult
 UDPSocketParent::RecvLeaveMulticast(const nsCString& aMulticastAddress,
                                     const nsCString& aInterface)
 {
+  if (!mSocket) {
+    NS_WARNING("multicast socket is closed");
+    FireInternalError(__LINE__);
+    return IPC_OK();
+  }
+
   nsresult rv = mSocket->LeaveMulticast(aMulticastAddress, aInterface);
 
   if (NS_WARN_IF(NS_FAILED(rv))) {
